File: CS/PersonalOrganizer/ContactContextDataModel/Runtime/ContactRepository.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using PersonalOrganizer.Common.Utils;
using PersonalOrganizer.Model;
using PersonalOrganizer.Common.DataModel;
using PersonalOrganizer.Common.DataModel.EntityFramework;

namespace PersonalOrganizer.ContactContextDataModel {
    public class ContactRepository : DbRepository<Contact, int, ContactContext>, IContactRepository {
        public ContactRepository(DbUnitOfWork<ContactContext> unitOfWork)
            : base(unitOfWork, context => context.Set<Contact>(), x => x.Id) {
        }
    }
}

File: VB/PersonalOrganizer/ContactContextDataModel/Runtime/ContactRepository.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Common.Utils
Imports PersonalOrganizer.Model
Imports Common.DataModel
Imports Common.DataModel.EntityFramework
Namespace ContactContextDataModel
    Public Class ContactRepository
        Inherits DbRepository(Of Contact, Integer, ContactContext)
        Implements IContactRepository
        Public Sub New(ByVal unitOfWork As DbUnitOfWork(Of ContactContext))
            MyBase.New(unitOfWork, Function(context) context.[Set](Of Contact)(), Function(x) x.Id)
        End Sub
    End Class
End Namespace
